orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java

```diff
@@ -189,18 +416,93 @@ public boolean isStoredInEntityStructure(AssociationKeyMetadata associationKeyMe
 
 	@Override
 	public Number nextValue(NextValueRequest request) {
-		log.info( "NextValueRequest:" + request + "; " );
-		return ORecordId.EMPTY_RECORD_ID.getClusterPosition();
+		log.debug( "NextValueRequest:" + request + "; " );
+		Number nextValue = null;
+		IdSourceType type = request.getKey().getMetadata().getType();
+		if ( IdSourceType.SEQUENCE.equals( type ) ) {
+			String seqName = request.getKey().getMetadata().getName();
+			nextValue = SequenceUtil.getSequence( provider.getConnection(), seqName );
+		}
+		else {
+			throw new UnsupportedOperationException( "nextValue Not supported yet." );
+		}
+		return nextValue;
 	}
 
 	@Override
 	public boolean supportsSequences() {
-		return super.supportsSequences(); // To change body of generated methods, choose Tools | Templates.
+		return true;
 	}
 
 	@Override
-	public void forEachTuple(ModelConsumer consumer, EntityKeyMetadata... entityKeyMetadatas) {
-		throw new UnsupportedOperationException( "Not supported yet." );
+	public ClosableIterator<Tuple> executeBackendQuery(BackendQuery<String> backendQuery, QueryParameters queryParameters) {
+
+		Map<String, Object> parameters = getNamedParameterValuesConvertedByGridType( queryParameters );
+		String nativeQuery = buildNativeQuery( backendQuery, queryParameters );
+		try {
+			PreparedStatement pstmt = provider.getConnection().prepareStatement( nativeQuery );
+			int paramIndex = 1;
+			for ( Map.Entry<String, TypedGridValue> entry : queryParameters.getNamedParameters().entrySet() ) {
+				String key = entry.getKey();
+				TypedGridValue value = entry.getValue();
+				log.debug( "key: " + key + "; type:" + value.getType().getName() + "; value:" + value.getValue() );
+				try {
+					// @todo move to Map
+					if ( value.getType().getName().equals( "string" ) ) {
+						pstmt.setString( 1, (String) value.getValue() );
+					}
+					else if ( value.getType().getName().equals( "long" ) ) {
+						pstmt.setLong( 1, (Long) value.getValue() );
+					}
+				}
+				catch (SQLException sqle) {
+					throw log.cannotSetValueForParameter( paramIndex, sqle );
+				}
+				paramIndex++;
+			}
+			ResultSet rs = pstmt.executeQuery();
+			return new ResultSetTupleIterator( rs );
+		}
+		catch (SQLException e) {
+			throw log.cannotExecuteQuery( nativeQuery, e );
+		}
+	}
+
+	private String buildNativeQuery(BackendQuery<String> customQuery, QueryParameters queryParameters) {
+		return customQuery.getQuery();
+	}
+
+	/**
+	 * Returns a map with the named parameter values from the given parameters object, converted by the {@link GridType}
+	 * corresponding to each parameter type.
+	 */
+	private Map<String, Object> getNamedParameterValuesConvertedByGridType(QueryParameters queryParameters) {
+		log.debug( "getNamedParameterValuesConvertedByGridType. named parameters: " + queryParameters.getNamedParameters().size() );
```

Comment:
It would be better to use the debugf for these logs, this way you can aviod string concatanation when the log is not required